File: game_server/GameRes/src/NPGameRes/GameObjs/PlayerVariable/PlayerVariableObj/NPPlayerVariable_CS_MARS_BUILDING_LVL.java
package NPGameRes.GameObjs.PlayerVariable.PlayerVariableObj;

import ALServerLog.ALServerLog;
import NPCommon.CommonObj.NPStringReader;
import NPEnum.ENPPlayerVariableType;
import NPGameRes.GameObjs.PlayerVariable._ANPBasicPlayerVariableObj;

public class NPPlayerVariable_CS_MARS_BUILDING_LVL extends _ANPBasicPlayerVariableObj
{
    private long _m_lId;

    public long getId()
    {
        return _m_lId;
    }
    
    /******************
     * 获取条件类型
     */
    @Override
    public ENPPlayerVariableType variableType()
    {
        return ENPPlayerVariableType.CS_MARS_BUILDING_LVL;
    }

    public static NPPlayerVariable_CS_MARS_BUILDING_LVL readVariable(NPStringReader _reader)
    {
        NPPlayerVariable_CS_MARS_BUILDING_LVL variableObj = new NPPlayerVariable_CS_MARS_BUILDING_LVL();

        try
        {
            String paramS = _reader.readItem('@');

            variableObj._m_lId = Integer.valueOf(paramS);

            return variableObj;
        } 
        catch (Exception e)
        {
            ALServerLog.Error("高级公式配置错误 - Value example: CS_MARS_BUILDING_LVL Error Str: " + _reader.getSrcString());
            return null;
        }
    }
}

File: game_client/Assets/Scripts/ResScripts/NPResCommon/PlayerVariable/NPPlayerVariableObj/NPPlayerVariable_CS_MARS_BUILDING_LVL.cs
using ALPackage;
using NPEnum;
using System;


namespace GOE
{
    /// <summary>
    /// 指定火星建筑等级 CS_MARS_BUILDING_LVL@火星建筑ID
    /// </summary>
    public class NPPlayerVariable_CS_MARS_BUILDING_LVL : _ANPBasicPlayerVariableObj
    {
        private long _m_lBuildingId;//火星建筑ID

        protected NPPlayerVariable_CS_MARS_BUILDING_LVL()
        {
        }

        /******************
       * 获取条件类型
       */
        public override ENPPlayerVariableType variableType { get { return ENPPlayerVariableType.CS_MARS_BUILDING_LVL; } }

        public override long calPlayerValue(NPVarInfo _variableInfo)
        {
            long level = 0;
#if NP_GAME
            MarsBuildingInfo buildingInfo = NPPlayer.instance.marsComp.buildingSubComponent.getBuildingInfoById(_m_lBuildingId);
            if (buildingInfo != null && buildingInfo.state != MarsBuildingInfo.StateType.Unbuilt && buildingInfo.state != MarsBuildingInfo.StateType.Constructing)
                level = buildingInfo.level;
#endif
            return level;
        }

        public static NPPlayerVariable_CS_MARS_BUILDING_LVL readVariable(ALStringReader _reader)
        {
            NPPlayerVariable_CS_MARS_BUILDING_LVL variableObj = new NPPlayerVariable_CS_MARS_BUILDING_LVL();
            string buildingIdStr = _reader.readItem('@');

            if (string.IsNullOrEmpty(buildingIdStr))
            {
                UnityEngine.Debug.LogError("高级公式——指定火星建筑等级 CS_MARS_BUILDING_LVL@火星建筑ID Error Str: " + _reader.srcString);
                return null;
            }

            try
            {
                variableObj._m_lBuildingId = long.Parse(buildingIdStr);
                return variableObj;
            }
            catch (Exception)
            {
                UnityEngine.Debug.LogError("高级公式——指定火星建筑等级 CS_MARS_BUILDING_LVL@火星建筑ID Error Str: " + _reader.srcString);
                return null;
            }
        }
    }
}
